internal/controllers/inventory_controller.go

```diff
@@ -371,10 +374,156 @@ func (t *reconcilerTask) setupAlarmSubscriptionServerConfig(ctx context.Context,
 	return
 }
 
+// setupOAuthClient is a wrapper around the similarly named method in the utils package.  The purpose for this wrapper
+// is to gather the parameters required by the utils version of this function before invoking it.  The reason for the
+// two layers is that it is expected that other parts of the system will need to get these inputs parameters differently
+// (i.e., from mounted files rather than API objects) but will need to share the underlying client creation code.
+func (t *reconcilerTask) setupOAuthClient(ctx context.Context) (*http.Client, error) {
+	var caBundle string
+	if t.object.Spec.CaBundleName != nil {
+		cm, err := utils.GetConfigmap(ctx, t.client, *t.object.Spec.CaBundleName, t.object.Namespace)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get configmap: %s", err.Error())
+		}
+
+		caBundle, err = utils.GetConfigMapField(cm, "ca-bundle.pem")
+		if err != nil {
+			return nil, fmt.Errorf("failed to get certificate bundle from configmap: %s", err.Error())
+		}
+	}
+
+	config := utils.OAuthClientConfig{
+		CaBundle: []byte(caBundle),
+	}
+
+	authConfig := t.object.Spec.SmoConfig.OAuthConfig
```

Comment:
Do you need a nil check for `SmoConfig` here, or just relying on the caller of the function to have already checked?

Reply:
Checked by the caller.  It wouldn't invoke any of this if it wasn't configured.